airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -271,6 +273,22 @@ def __init__(  # pylint: disable=too-many-arguments,too-many-locals
         self.client: CoreV1Api = None
         self.pod: k8s.V1Pod = None
 
+    def add_template_fields_to_env_vars(self, env_vars):
```

Comment:
Also this only address Env Variables, what about other template_field

Reply:
https://github.com/apache/airflow/blob/3909232fafd09ac72b49010ecdfd6ea48f06d5cf/airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py#L158-L166

Reply:
We should have something generic that applies to all of this in KPO

Reply:
@kaxil this is specific to the KPO as the k8s.V1EnvVar object doesn't have a templated_fields option. This problem only applies to fields that take in k8s objects. Do you have thoughts on other things that are the same?

The thing you are pointing out says that the FIELD is templatable, but if the object within that field is a class that doesn't have a templated_fields field, then the render template function doesn't know what to do. All other elements in that list are strings or dictionaries so they shouldn't be an issue.

Reply:
right that's correct